daffodil-runtime1-unparser/src/main/scala/edu/illinois/ncsa/daffodil/processors/unparsers/PackedBinaryUnparserTraits.scala

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package edu.illinois.ncsa.daffodil.processors.unparsers
+
+import edu.illinois.ncsa.daffodil.processors.ElementRuntimeData
+import edu.illinois.ncsa.daffodil.processors.ParseOrUnparseState
+import edu.illinois.ncsa.daffodil.util.Maybe._
+import java.lang.{ Number => JNumber }
+import java.math.{ BigDecimal => JBigDecimal }
+import edu.illinois.ncsa.daffodil.exceptions.Assert
+import edu.illinois.ncsa.daffodil.io.DataOutputStream
+import edu.illinois.ncsa.daffodil.io.FormatInfo
+import edu.illinois.ncsa.daffodil.schema.annotation.props.gen.BinaryNumberCheckPolicy
+
+trait PackedBinaryConversion {
+  def fromBigInteger(num: String, nBits: Int, signCodes: String, policy: BinaryNumberCheckPolicy): Array[Byte]
+}
+
+abstract class PackedBinaryBaseUnparser(
+  e: ElementRuntimeData,
+  binaryPackedSignCodes: String = "",
```

Comment:
What does a empty string binaryPackedSignCodes mean? Shouldn't there always be sign codes? Maybe the default value can be removed? Same as below, are there cases when a binaryNumberCheckPolicy isn't provided? Can the default be removed for that too?

Reply:
So this may just be my ignorance of Scala, but I was trying to provide a generic base class that could be used by Packed, BCD, and IBM4690.  The binaryPackedSignCodes and binaryNumberCheckPolicy properties are only used by PackedDecimal, so I thought the best way to make this generic interface work would be to have default values which are simply ignored by BCD and IBM4690.

Reply:
Gotcha. That's probably not ideal. I think you can change fromBigInteger to not accept a signCode and policy, so the definition just looks like this:
```
def fromBigInteger(num: String, nBits: Int): Array[Byte]
```
The IBM Packed would implement it like so:
```
override def fromBigInteger(num: String, nBits: Int): Array[Byte] = IBM4690FromBigInteger(num, nBits)
```
Likewise for BCD.

For PackedNumberBaseUnparser, you already pass in the sign code and policy (should be changed to the new data structure mentioned elsewhere) to its constructor, so it would just look like this:
```
override def fromBigInteger(num: String, nBits: Int): Array[Byte] = packedFromBigInteger(num, nBits, binaryPackedSignCodes, binaryNumberCheckPolicy)
```
With this, I think PackedBinaryBaseUnparser can just call fromBigInteger(num, nBits) and the implementations of fromBigInteger have any extra information they might need, so you can remove those two variables from the PackedBinaryBaseUnparser class and don't have to worry about defaults.